File: src/BukkitPE/inventory/Recipe.php
<?php
namespace BukkitPE\inventory;

use BukkitPE\utils\UUID;

interface Recipe{

	/**
	 * @return \BukkitPE\item\Item
	 */
	public function getResult();

	public function registerToCraftingManager();

	/**
	 * @return UUID
	 */
	public function getId();
}

File: src/main/java/net/BukkitPE/inventory/Recipe.java
package net.BukkitPE.inventory;

import net.BukkitPE.item.Item;

import java.util.UUID;

/**
 * BukkitPE Project
 */
public interface Recipe {

    Item getResult();

    void registerToCraftingManager();

    UUID getId();

    void setId(UUID id);
}
